src/mount.aznfs

```diff
@@ -0,0 +1,352 @@
+                continue
+            fi
+
+            vecho "Using local IP ($local_ip) for aznfs." 1>&2 
+            break 
+        done
+
+        if [ $_4thoctet -eq 255 ]; then 
+            continue
+        fi
+
+        # Happy path!
+        LOCAL_IP=$local_ip
+        return 0
+    done
+
+    # If we come here we didn't get a free address to use.
+    return 1
+}
+
+#
+# Ensure aznfswatchdog service is running, if not bail out with an appropriate
+# error.
+#
+# TODO: Make sure this works on all supported distros.  
+#
+ensure_aznfswatchdog()
+{
+    is_aznfswatchdog_active=$(systemctl is-active --quiet aznfswatchdog) 
+    if [[ $is_aznfswatchdog_active -ne 0 ]]; then
+        eecho "aznfswatchdog service not running!"
+        pecho "Start the aznfswatchdog service using 'systemctl start aznfswatchdog' and try again."
+        pecho "If the problem persists, contact Microsoft."
+        exit 1
+    fi
+}
+
+# [account.blob.core.windows.net:/account/container /mnt/aznfs -o rw,tcp,nolock,nconnect=16] 
+vecho "[$(date -u)] Got arguments: [$*]"
+
+# Check if aznfswatchdog service is running.
+ensure_aznfswatchdog
+
+# MOUNTMAP file must have been created by aznfswatchdog service.
+if [ ! -f $MOUNTMAP ]; then
+    eecho "[FATAL] ${MOUNTMAP} not found!"
+    pecho "Start the aznfswatchdog service using 'systemctl start aznfswatchdog' and try again."
+    pecho "If the problem persists, contact Microsoft."
+    exit 1
+fi
+
+nfs_host=$(get_host_from_share "$1") 
+
+if ! is_valid_blob_fqdn $nfs_host; then 
+    eecho "Not a valid Azure Blob NFS endpoint: ${nfs_host}!"
+    eecho "Must be of the form 'account.blob.core.windows.net'."
+    exit 1 
+fi
+
+nfs_ip=$(resolve_ipv4 ${nfs_host})
+
+if [ -z "$nfs_ip" ]; then 
+    eecho "Cannot resolve IP address for ${nfs_host}!" 
+    exit 1 
+fi
+
+#
+# If user is using private endpoint then we can directly mount without the help
+# of aznfswatchdog service.
+#
+if is_private_ip $nfs_ip; then
+    vecho "Host IP is private ($nfs_ip), using direct NFS mount."
+    mount -t nfs $*
+    exit
+fi
+
+nfs_dir=$(get_dir_from_share "$1") 
+
+if [ -z "$nfs_dir" ]; then
+    eecho "Bad share name: ${1}."
+    eecho "Share to be mounted must be of the form 'account.blob.core.windows.net:/account/container'." 
+    exit 1 
+fi 
+
+mount_point="$2" 
+mount_options="$4" 
+ 
+get_free_local_ip
+
+if [ $? -ne 0 ]; then
+    eecho "Could not find a free local IP to use for aznfs using DEFAULT_AZNFS_IP_PREFIXES=${DEFAULT_AZNFS_IP_PREFIXES}!"
+    eecho "Set AZNFS_IP_PREFIXES env variable correctly to provide free addresses for use by aznfs!"
+    exit 1
+fi
+
+vecho "nfs_host=[$nfs_host]"
+vecho "nfs_ip=[$nfs_ip]"
+vecho "nfs_dir=[$nfs_dir]"
+vecho "mount_point=[$mount_point]"
+vecho "mount_options=[$mount_options]"
+vecho "local_ip=$LOCAL_IP"
+
+# Check if DNAT rule for forwarding LOCAL_IP traffic to the actual blob endpoint IP already exists.
```

Comment:
this comment is misleading.. we don't expect it to already exist.
You should say 
Add DNAT rule for forwarding LOCAL_IP traffic to the actual blob endpoint IP address